readthedocs/proxito/tests/test_middleware.py

```diff
@@ -119,20 +126,16 @@ def test_canonical_cname_redirect_public_domain(self):
         cname = 'docs.random.com'
         domain = get(Domain, project=self.pip, domain=cname, canonical=False, https=False)
 
-        request = self.request(method='get', path=self.url, HTTP_HOST=cname)
-        res = self.run_middleware(request)
-        self.assertIsNone(res)
-        self.assertTrue(hasattr(request, 'canonicalize'))
-        self.assertEqual(request.canonicalize, 'noncanonical-cname')
+        resp = self.client.get(self.url, HTTP_HOST=cname)
+        self.assertEqual(resp.status_code, 302)
+        self.assertEqual(resp["X-RTD-Redirect"], "noncanonical-cname")
```

Comment:
Shouldn't we create a `RedirectType.to_non_canonical_domain` and use it here?

Reply:
We aren't setting that value from anywhere, this test is marked as fail too. This looks like a feature we wanted to implement at some point, I'm +1 on just deleting these tests.

Reply:
> Requests to a custom domain should redirect to the public domain or canonical domain if not canonical.

I guess that's when users don't have any custom domain set as `canonical` we want to redirect to the public domain?

@ericholscher may have more information about this "lost feature"

Reply:
https://github.com/readthedocs/readthedocs.org/pull/6905#issuecomment-614838263 I think is the conversation. I think it's mostly just awkward to redirect people away from custom domains. There are use cases where people might want to have docs on a custom domain, but not have it be canonical. 

So I'm +1 on deleting this I think.